src/kibana/components/vislib/components/color/color.js

```diff
@@ -9,9 +9,10 @@ define(function (require) {
      * a lookup table that associates the values (key) with a hex color (value).
      * Returns a function that accepts a value (i.e. a string or number)
      * and returns a hex color associated with that value.
+     * Allows an empty value to match to the first color.
      */
 
-    return function (arrayOfStringsOrNumbers) {
+    return function (arrayOfStringsOrNumbers, matchEmptyToFirst) {
```

Comment:
When you have a bar chart that only has one series, the individual data points don't have a label.  So it originally worked by generating a fake label just for the legend (https://github.com/elastic/kibana/blob/master/src/kibana/components/vislib/lib/data.js#L45) and creating two color functions: one for the legend (https://github.com/elastic/kibana/blob/master/src/kibana/components/vislib/lib/data.js#L53) and one for the actual data (https://github.com/elastic/kibana/blob/master/src/kibana/components/vislib/lib/data.js#L53).

So the legend has a color function that uses an array with a single string value and the data points have a color function that uses an array with a single empty string value.

I thought about adding a label to all the data points to match the legend but wasn't sure that you would want that extra work

Another thought was to modify where it grabbed the label (https://github.com/elastic/kibana/blob/master/src/kibana/components/vislib/lib/data.js#L53) and if it is blank, set it to the 'yAxisLabel'.

Would you prefer either of those to my current code?  Do you have another suggestion?